scripts/Jenkinsfile

```diff
@@ -137,19 +119,47 @@ pipeline {
                         }
                     }
                 }
-                stage('Package and upload helm charts') {
-                    steps {
-                        script {
-                            cicdLibrary.uploadHelmCharts(env.pathToCharts)
-                        }
-                    }
+            }
+        }
+
+        stage("Build and upload notebooks Docker images") {
+            steps {
+                script {
+                    def buildArgs = """--build-arg NPM_EXTRA_INDEX_URL=https://${env.param_npm_private_url}
+                                       --build-arg ODAHU_PLUGIN_VERSION=${Globals.buildVersion}
+                                    """
+                    cicdLibrary.buildDockerImage(
+                        'base-notebook',
+                        '.',
+                        'containers/jupyter-stacks/Dockerfile',
+                        "${buildArgs} --build-arg NOTEBOOK_TYPE=base".replaceAll("\\s+"," ")
```

Comment:
What is a reason to use the `replaceAll` function?

Reply:
Jenkins job was unable to pass multiline string as single variable, so job was failing with
```
"docker build" requires exactly 1 argument.
```
(see build 7 of `Build_OdahuFlowJupyterlabPlugin_Test_JHub` job)

Reply:
```groovy
                 script {
                    def buildArgs = """--build-arg NPM_EXTRA_INDEX_URL=https://${env.param_npm_private_url}
                                       --build-arg ODAHU_PLUGIN_VERSION=${Globals.buildVersion}
                                    """
                    cicdLibrary.buildDockerImage(
                        'base-notebook',
                        '.',
                        'containers/jupyter-stacks/Dockerfile',
                        "${buildArgs} --build-arg NOTEBOOK_TYPE=base".replaceAll("\\s+"," ")
                    )
                    cicdLibrary.buildDockerImage(
                        'datascience-notebook',
                        '.',
                        'containers/jupyter-stacks/Dockerfile',
                        "${buildArgs} --build-arg NOTEBOOK_TYPE=datascience".replaceAll("\\s+"," ")
                    )
                    cicdLibrary.buildDockerImage(
                        'tensorflow-notebook',
                        '.',
                        'containers/jupyter-stacks/Dockerfile',
                        "${buildArgs} --build-arg NOTEBOOK_TYPE=tensorflow".replaceAll("\\s+"," ")
                    )
                }
```
`->`
```groovy
                 script {
                    def buildArgs = """--build-arg NPM_EXTRA_INDEX_URL=https://${env.param_npm_private_url}
                                       --build-arg ODAHU_PLUGIN_VERSION=${Globals.buildVersion}
                                    """.replaceAll("\\s+"," ")
                    cicdLibrary.buildDockerImage(
                        'base-notebook',
                        '.',
                        'containers/jupyter-stacks/Dockerfile',
                        "${buildArgs} --build-arg NOTEBOOK_TYPE=base"
                    )
                    cicdLibrary.buildDockerImage(
                        'datascience-notebook',
                        '.',
                        'containers/jupyter-stacks/Dockerfile',
                        "${buildArgs} --build-arg NOTEBOOK_TYPE=datascience"
                    )
                    cicdLibrary.buildDockerImage(
                        'tensorflow-notebook',
                        '.',
                        'containers/jupyter-stacks/Dockerfile',
                        "${buildArgs} --build-arg NOTEBOOK_TYPE=tensorflow"
                    )
                }
```

Reply:
yup, already did in 2f1e4dc38b894d110c67c4852b69c6664b4d1955